useragent/src/main/java/io/github/ackeecz/useragent/UserAgent.java

```diff
@@ -1,28 +1,35 @@
 package io.github.ackeecz.useragent;
 
 import android.content.Context;
-import android.content.pm.ApplicationInfo;
-import android.content.pm.PackageManager;
-import android.os.Build;
 
-import java.text.Normalizer;
-import java.util.Locale;
-import java.util.regex.Pattern;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import io.github.ackeecz.useragent.clientinfo.AndroidClientInfo;
+import io.github.ackeecz.useragent.normalizer.DiacriticsNormalizer;
+import io.github.ackeecz.useragent.normalizer.Normalizer;
 
 /**
- * Singleton class that generates user agent string with app specific data for easier server device distinction
+ * Class that generates user agent string with app specific data for easier server device distinction.
  */
 public class UserAgent {
```

Comment:
Why don't you convert this class to kotlin if the rest is converted anyway? For backward compatibility? You could still probably achieve it with kotlin + some java interop annotations, right?

Reply:
I was too lazy to deal with it, but ok :D f0adf6585f01b8c7e5642f57cf1375cfaf315ac1